tool/tctl/common/autoupdate_command.go

```diff
@@ -0,0 +1,206 @@
+	c.watchCmd.Flag("proxy", `URL of the proxy`).StringVar(&c.proxy)
+}
+
+// TryRun takes the CLI command as an argument and executes it.
+func (c *AutoupdateCommand) TryRun(ctx context.Context, cmd string, client *authclient.Client) (match bool, err error) {
+	switch cmd {
+	case c.updateCmd.FullCommand():
+		err = c.Upsert(ctx, client)
+	case c.getCmd.FullCommand():
+		err = c.Get(ctx, client)
+	case c.watchCmd.FullCommand():
+		err = c.Watch(ctx, client)
+	default:
+		return false, nil
+	}
+	return true, trace.Wrap(err)
+}
+
+// Upsert works with autoupdate_config and autoupdate_version resources to create or update
+func (c *AutoupdateCommand) Upsert(ctx context.Context, client *authclient.Client) error {
+	serviceClient := client.AutoupdateServiceClient()
+
+	if c.toolsAutoupdate != "" {
+		config, err := serviceClient.GetAutoupdateConfig(ctx, &autoupdate.GetAutoupdateConfigRequest{})
+		if trace.IsNotFound(err) {
+			if config, err = update.NewAutoupdateConfig(&autoupdate.AutoupdateConfigSpec{}); err != nil {
+				return trace.Wrap(err)
+			}
+		} else if err != nil {
+			return trace.Wrap(err)
+		}
+		isEnabled := c.toolsAutoupdate == "on"
+		if isEnabled != config.Spec.ToolsAutoupdate {
+			config.Spec.ToolsAutoupdate = isEnabled
+			if _, err := serviceClient.UpsertAutoupdateConfig(ctx, &autoupdate.UpsertAutoupdateConfigRequest{
+				Config: config,
+			}); err != nil {
+				return trace.Wrap(err)
+			}
+			fmt.Println("autoupdate_config has been upserted")
+		}
+	}
+
+	version, err := client.AutoupdateServiceClient().GetAutoupdateVersion(ctx, &autoupdate.GetAutoupdateVersionRequest{})
+	if trace.IsNotFound(err) {
+		if version, err = update.NewAutoupdateVersion(&autoupdate.AutoupdateVersionSpec{}); err != nil {
+			return trace.Wrap(err)
+		}
+	} else if err != nil {
+		return trace.Wrap(err)
+	}
+	if version.Spec.ToolsVersion != c.toolsAutoupdateVersion {
+		version.Spec.ToolsVersion = c.toolsAutoupdateVersion
+		if _, err := serviceClient.UpsertAutoupdateVersion(ctx, &autoupdate.UpsertAutoupdateVersionRequest{
+			Version: version,
+		}); err != nil {
+			return trace.Wrap(err)
+		}
+		fmt.Println("autoupdate_version has been upserted")
+	}
+
+	return nil
+}
+
+// Get makes request to fetch tools autoupdate version, if proxy flag is not set
+// authorized handler should be used.
+func (c *AutoupdateCommand) Get(ctx context.Context, client *authclient.Client) error {
+	response, err := c.get(ctx, client)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := utils.WriteJSON(os.Stdout, response); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// Watch launch the watcher of the tools autoupdate version updates.
+func (c *AutoupdateCommand) Watch(ctx context.Context, client *authclient.Client) error {
+	current := teleport.SemVersion
+	ticker := interval.New(interval.Config{
+		Duration:      time.Minute,
+		FirstDuration: time.Second,
```

Comment:
Curious why 1s instead of 0?

Reply:
```go
	firstDuration := cfg.FirstDuration
	if firstDuration == 0 {
		firstDuration = interval.duration()
	}
```
it is going to be ignored if there are 0, made similar to some places where interval lib is used
I would probably change to this one:

```go
	ticker := interval.New(interval.Config{
		Duration:      0,
		FirstDuration: time.Second,
	})
	ticker.FireNow()
	defer ticker.Stop()
```
also I forgot to add `defer ticker.Stop()`

Reply:
Also option to structure it like this:
```golang
for {
  c.get()
  case ...
}